pkg/identity/key/global_identity.go

```diff
@@ -19,20 +20,29 @@ const (
 )
 
 // GlobalIdentity is the structure used to store an identity
+// Note: GlobalIdentity is immutable; use the mutator methods.
 type GlobalIdentity struct {
 	labels.LabelArray
```

Comment:
optional nit: Doing a quick usage check of `GlobalIdentity.LabelArray` I see ~30 only usages, most of them are in tests and/or are value constructions.

Might be worth making the field private so the immutability is a bit more strongly enforced.